packages/website/docs/_samples/main/Menu/DynamicallyAddedItems/main.js

```diff
@@ -0,0 +1,64 @@
+import "@ui5/webcomponents/dist/Menu.js";
+import "@ui5/webcomponents/dist/MenuItem.js";
+
+import "@ui5/webcomponents-icons/dist/add-document.js";
+import "@ui5/webcomponents-icons/dist/add-folder.js";
+import "@ui5/webcomponents-icons/dist/open-folder.js";
+import "@ui5/webcomponents-icons/dist/save.js";
+import "@ui5/webcomponents-icons/dist/upload-to-cloud.js";
+import "@ui5/webcomponents-icons/dist/action-settings.js";
+import "@ui5/webcomponents-icons/dist/journey-arrive.js";
+import "@ui5/webcomponents-icons/dist/slim-arrow-down.js";
+
+const btnOpenBasic = document.getElementById("btnOpenBasic");
+const menuSubs = document.getElementById("menuSubs");
+const btnAddOpenerDelay = document.getElementById("btnAddOpenerDelay");
+const delaymenu = document.getElementById("delaymenu");
+
+
+btnAddOpenerDelay.addEventListener("click", function() {
+    delaymenu.opener = "btnAddOpenerDelay";
+    delaymenu.open = !delaymenu.open;
+});
+
+btnOpenBasic.addEventListener("click", function() {
+    menuSubs.opener = "btnOpenBasic";
+    menuSubs.open = !menuSubs.open;
+});
+
+let menuItemsLoaded = false;
+
+delaymenu.addEventListener("ui5-before-open", function(event) {
+    if (!menuItemsLoaded) {
+        setTimeout(function() {
+            delaymenu.removeAttribute("loading");
+            delaymenu.removeAttribute("loading-delay");
```

Comment:
You can reuse the function the function for both event listeners event target is the element where items should be appended. 
You can use loading property as a flag instead of creating a new variable.
Use property assignment instead of removeAttribute method
For naming variables use camelCase